// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTCashEnums";
option optimize_for = LITE_RUNTIME;

enum CashType {
    CASHTYPE_ERROR = 0;
    CASHTYPE_LUCRE = 1;
}

enum TokenState {
    TOKENSTATE_ERROR = 0;
    TOKENSTATE_BLINDED = 1;
    TOKENSTATE_SIGNED = 2;
    TOKENSTATE_READY = 3;
    TOKENSTATE_SPENT = 4;
    TOKENSTATE_EXPIRED = 5;
}

enum PurseType {
    PURSETYPE_ERROR = 0;
    PURSETYPE_REQUEST = 1;
    PURSETYPE_ISSUE = 2;
    PURSETYPE_NORMAL = 3;
}

